From ea43d1e039be28e20ec313c3861c4bb54c189c16 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Mon, 06 Feb 2023 09:21:19 +0200
Subject: [PATCH] [SW-115382] ppboot: pll: use existing pll APIs

SRAM PLLs must be configured before running BIST operations.
Currently gaudi2_sram_pll_cfg is used to configure the plls is duplicated
in gaudi2 specific pll.c file.

Removed the duplicate code and calling existing hl_dev_pll_init API

Change-Id: Idd7f0012f1aae62d4287d72a91d4fa6a1cd372aa
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/arch/gaudi2/common/pll.c b/arch/gaudi2/common/pll.c
index 48373ed..80de124 100644
--- a/arch/gaudi2/common/pll.c
+++ b/arch/gaudi2/common/pll.c
@@ -34,105 +34,6 @@
 	}
 };
 
-static struct hl_pll_cfg gaudi2_sram_pll_cfg[] = {
-	{
-		.name = "PCI_PLL",
-		.base = mmDCORE0_PCI_PLL_CTRL_BASE,
-		.nr = 1, .nf = 60, .od = 3, .nb = 2,	/* 500 */
-		/*
-		 * div 0 div_factor differ from spec due to CDC workaround required for
-		 * pcie link init. it is set to nominal clk by preboot after the pcie link
-		 * is up.
-		 */
-		.div_factor = { 4, -1, 4, 4 },
-		.div_sel = { HL_PLL_DIV_PLL,
-			     HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "DCORE0_IF",
-		.base = mmDCORE0_XBAR_IF_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
-		.div_factor = { 0, 1, 1, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "DCORE1_IF",
-		.base = mmDCORE1_XBAR_IF_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
-		.div_factor = { 0, 1, 1, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "DCORE2_IF",
-		.base = mmDCORE2_XBAR_IF_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
-		.div_factor = { 0, 1, 1, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "DCORE3_IF",
-		.base = mmDCORE3_XBAR_IF_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
-		.div_factor = { 0, 1, 1, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "DCORE0_MESH",
-		.base = mmDCORE0_XBAR_MESH_PLL_CTRL_BASE,
-		.nr = 1, .nf = 34, .od = 1, .nb = 1,	/* 1700 */
-		.div_factor = { 0, 1, 2, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "DCORE1_MESH",
-		.base = mmDCORE1_XBAR_MESH_PLL_CTRL_BASE,
-		.nr = 1, .nf = 34, .od = 1, .nb = 1,	/* 1700 */
-		.div_factor = { 0, 1, 2, 7 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL,
-			     HL_PLL_DIV_PLL
-		},
-	}, {
-		.name = "DCORE2_SRAM",
-		.base = mmDCORE2_XBAR_BANK_PLL_CTRL_BASE,
-		.nr = 1, .nf = 63, .od = 3, .nb = 3,	/* 350 */
-		.div_factor = { 0, -1, -1, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
-		},
-	}, {
-		.name = "DCORE3_SRAM",
-		.base = mmDCORE3_XBAR_BANK_PLL_CTRL_BASE,
-		.nr = 1, .nf = 63, .od = 3, .nb = 3,	/* 350 */
-		.div_factor = { 0, -1, -1, -1 },
-		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
-		},
-	}
-};
-
 void hl_pll_reset(struct hl_pll_cfg *cfg, bool reset)
 {
 	u32 reg = regmap_readl(cfg->base + GAUDI2_PLL_CFG);
@@ -202,5 +103,5 @@
 
 void ppboot_plls_init(void)
 {
-	hl_pll_init(gaudi2_sram_pll_cfg, ARRAY_SIZE(gaudi2_sram_pll_cfg));
+	hl_dev_pll_init();
 }
